contrib/test.sh

```diff
@@ -2,14 +2,36 @@
 
 set -ex
 
-CRATES="bitcoin"
+if [ -n "$1" ];
+then
+	deps="$1"
+else
+	deps="minimal recent"
+fi
 
-for crate in ${CRATES}
+CRATES="bitcoin internals"
+
+for dep in $deps
 do
-    (
-        cd "$crate"
-        ./contrib/test.sh
-    )
+	cp Cargo-$dep.lock Cargo.lock
+	for crate in ${CRATES}
+	do
+	    (
+		cd "$crate"
+		./contrib/test.sh
+	    )
+	done
+	if [ $dep = recent ];
+	then
+		# We always test committed dependencies but we want to warn if they could've been updated
+		cargo update
+		if diff Cargo-recent.lock Cargo.lock;
+		then
+			echo Dependencies are up to date
+		else
+			echo "::warning file=Cargo-recent.lock::Dependencies could be updated"
```

Comment:
Can I ask, why the double colons in `::warning`? We could use colors
```
echo "\e[33mWarning:\e[0m ..."
```

Reply:
This is a special signal for GitHub to show the warning at that file. Sadly one will only see it when explicitly looking at the file but that's still better than looking into logs.

Reply:
Oh cool, cheers - I've never seen that before.

Reply:
Me neither, I just thought "is there a way to warn in GH actions?" and succeeded googling it. :)